pkg/cloud/aws/sqs/service.go

```diff
@@ -0,0 +1,282 @@
+	}
+
+	if exists {
+		return s.GetPropertiesByName(ctx, settings.QueueName)
+	}
+
+	if !exists && !s.settings.AutoCreate {
+		return nil, fmt.Errorf("sqs queue with name %s does not exist", settings.QueueName)
+	}
+
+	attributes, err := s.createDeadLetterQueue(ctx, settings)
+	if err != nil {
+		return nil, err
+	}
+
+	sqsInput := &sqs.CreateQueueInput{
+		QueueName:  aws.String(settings.QueueName),
+		Attributes: make(map[string]string),
+	}
+
+	for k, v := range attributes {
+		sqsInput.Attributes[k] = v
+	}
+
+	if settings.Fifo.Enabled {
+		sqsInput.Attributes[string(types.QueueAttributeNameFifoQueue)] = "true"
+	}
+
+	if settings.Fifo.Enabled && settings.Fifo.ContentBasedDeduplication {
+		sqsInput.Attributes[string(types.QueueAttributeNameContentBasedDeduplication)] = "true"
+	}
+
+	props, err := s.doCreateQueue(ctx, sqsInput)
+	if err != nil {
+		return nil, err
+	}
+
+	visibilityTimeout := DefaultVisibilityTimeout
+	if settings.VisibilityTimeout > 0 {
+		visibilityTimeout = strconv.Itoa(settings.VisibilityTimeout)
+	}
+
+	_, err = s.client.SetQueueAttributes(ctx, &sqs.SetQueueAttributesInput{
+		QueueUrl: aws.String(props.Url),
+		Attributes: map[string]string{
+			string(types.QueueAttributeNameVisibilityTimeout): visibilityTimeout,
+		},
+	})
+
+	return props, err
+}
+
+func (s *service) QueueExists(ctx context.Context, name string) (bool, error) {
+	s.logger.WithFields(log.Fields{
+		"name": name,
+	}).Info("checking the existence of sqs queue")
+
+	var err error
+	var url string
+
+	if url, err = s.GetUrl(ctx, name); err != nil {
+		return false, fmt.Errorf("can not get url of queue: %w", err)
+	}
+
+	if len(url) > 0 {
+		s.logger.Info("found queue %s with url %s", name, url)
+		return true, nil
+	}
+
+	s.logger.WithFields(log.Fields{
+		"name": name,
+	}).Info("could not find queue")
+
+	return false, nil
+}
+
+func (s *service) GetPropertiesByArn(ctx context.Context, arn string) (*Properties, error) {
+	i := strings.LastIndex(arn, ":")
+	name := arn[i+1:]
+
+	var err error
+	var url string
+
+	if url, err = s.GetUrl(ctx, name); err != nil {
+		return nil, fmt.Errorf("can not get url: %w", err)
+	}
+
+	return &Properties{
+		Name: name,
+		Url:  url,
+		Arn:  arn,
+	}, nil
+}
+
+func (s *service) GetPropertiesByName(ctx context.Context, name string) (*Properties, error) {
+	url, err := s.GetUrl(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+
+	arn, err := s.GetArn(ctx, url)
+	if err != nil {
+		return nil, err
+	}
+
+	properties := &Properties{
+		Name: name,
+		Url:  url,
+		Arn:  arn,
+	}
+
+	return properties, nil
+}
+
+func (s *service) GetUrl(ctx context.Context, name string) (string, error) {
+	var err error
+	var out *sqs.GetQueueUrlOutput
+
+	input := &sqs.GetQueueUrlInput{
+		QueueName: aws.String(name),
+	}
+
+	if out, err = s.client.GetQueueUrl(ctx, input); err != nil {
+		var errQueueDoesNotExist *types.QueueDoesNotExist
+		if errors.As(err, &errQueueDoesNotExist) {
+			return "", nil
+		}
+
+		return "", nil
```

Comment:
Not quite sure if I like swallowing a queue-does-not-exist error, but for some other error we maybe should return it instead of swallowing it?

Reply:
Actually that's just a copy from the old code..  
I guess there was a reason for having it like this? :thinking:

Reply:
If that reason was laziness - we should change it ^^
I don't like swallowing errors, especially if we don't know why

Reply:
I approve, but as I'm not sure of the side affects I would not like to tackle it now but instead do it in a dedicated PR..  
For now I would keep it like it was..